lib/project/results-model.coffee

```diff
@@ -210,10 +210,12 @@ class ResultsModel
       filePathInsertedIndex = binaryIndex(@paths, filePath, stringCompare)
       @paths.splice(filePathInsertedIndex, 0, filePath)
 
-    @matchCount += result.matches.length
-
-    @results[filePath] = result
-    @emitter.emit 'did-add-result', {filePath, result, filePathInsertedIndex}
+    bufferPromise = atom.project.buildBuffer(filePath)
+    bufferPromise.then (buffer) =>
+      result.lines = buffer.lines
```

Comment:
I'm not sure if it's a good idea to pass all lines rather than just those that will be used.

I'm not on the Atom team or anything though, so they may be fine with it :)